go/cmd/dolt/commands/push.go

```diff
@@ -87,93 +93,212 @@ func (cmd PushCmd) Exec(ctx context.Context, commandStr string, args []string, d
-		err = actions.HandleInitRemoteStorageClientErr(opts.Remote.Name, opts.Remote.Url, err)
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+// constructInterpolatedDoltPushQuery generates the sql query necessary to call the DOLT_PUSH() function
+// Also interpolates this query to prevent sql injection.
+func constructInterpolatedDoltPushQuery(apr *argparser.ArgParseResults) (string, error) {
+	var params []interface{}
+	var args []string
+
+	if setUpstream := apr.Contains(cli.SetUpstreamFlag); setUpstream {
+		args = append(args, "'--set-upstream'")
+	}
+	if force := apr.Contains(cli.ForceFlag); force {
+		args = append(args, "'--force'")
+	}
+	for _, arg := range apr.Args {
+		args = append(args, "?")
+		params = append(params, arg)
 	}
 
-	tmpDir, err := dEnv.TempTableFilesDir()
+	query := fmt.Sprintf("call dolt_push(%s)", strings.Join(args, ", "))
+	interpolatedQuery, err := dbr.InterpolateForDialect(query, params, dialect.MySQL)
 	if err != nil {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+		return "", err
 	}
-	var verr errhand.VerboseError
-	err = actions.DoPush(ctx, dEnv.RepoStateReader(), dEnv.RepoStateWriter(), dEnv.DoltDB, remoteDB, tmpDir, opts, buildProgStarter(defaultLanguage), stopProgFuncs)
+
+	return interpolatedQuery, nil
+}
+
+// getDefaultRemote gets the name of the default remote.
+func getDefaultRemote(sqlCtx *sql.Context, queryist cli.Queryist) (string, error) {
+	rows, err := GetRowsForSql(queryist, sqlCtx, "select name from dolt_remotes")
 	if err != nil {
-		verr = printInfoForPushError(err, opts.Remote, opts.DestRef, opts.RemoteRef)
+		return "", err
+	}
+	if len(rows) == 0 {
+		return "", env.ErrNoRemote
+	}
+	if len(rows) == 1 {
+		return rows[0][0].(string), nil
+	}
+	for _, row := range rows {
+		if row[0].(string) == "origin" {
+			return "origin", nil
+		}
 	}
+	return "", env.ErrCantDetermineDefault
+}
+
+// processFetchSpecs takes a string of fetch specs and returns the destination ref and remote ref
+// Assumes the fetch specs look something like: ["refs/heads/*:refs/remotes/origin/*"]
+func processFetchSpecs(fetchSpecs string, branch string) (destRef, remoteRef string) {
+	// TODO: replace string parsing with better logic when more JSONDocument functionality is available
+	destAndRemoteRefs := strings.Split(fetchSpecs, ":")
+	destRef = destAndRemoteRefs[0]
+	destRef = strings.TrimPrefix(destRef, "[\"")
+	destRef = strings.ReplaceAll(destRef, "*", branch)
+
+	remoteRef = destAndRemoteRefs[1]
+	remoteRef = strings.TrimSuffix(remoteRef, "\"]")
+	remoteRef = strings.ReplaceAll(remoteRef, "*", branch)
 
-	if opts.SetUpstream {
-		err := dEnv.RepoState.Save(dEnv.FS)
+	return
+}
+
+// printPushMessage prints the appropriate message for the given push output
+func printPushMessage(rows []sql.Row) error {
+	var statusCode int64
+	if intCode, ok := rows[0][0].(int64); ok {
+		statusCode = intCode
+	} else if strCode, ok := rows[0][0].(string); ok {
```

Comment:
Are the types not agreeing between local and remote executions? I would think the string version would come back as an integer of some type - is it not?

Reply:
Yeah it seems like the remote execution is coming back as a string, it won't cast as an int.

Reply:
Can you add a comment to this effect? I don't know what the root cause is, and I'm not going to ask you to sort it out. But this will confuse future readers